satpy/composites/glm.py

```diff
@@ -0,0 +1,121 @@
+        channel_result = primary_data + highlight_factor * max_factor
+
+    See the ``highlight_channel`` option to control if the effect is additive,
+    subtractive, or not applied at all.
+
+    """
+
+    def __init__(self, name, min_highlight=0.0, max_highlight=10.0, max_factor=0.8,
+                 highlight_channel=(True, True, False, None), **kwargs):
+        """Initialize composite with highlight factor options.
+
+        Args:
+            min_highlight (float): Minimum raw value of the "highlight" data
+                that will be used for linearly scaling the data along with
+                ``max_hightlight``.
+            max_highlight (float): Maximum raw value of the "highlight" data
+                that will be used for linearly scaling the data along with
+                ``min_hightlight``.
+            max_factor (float): Maximum effect that the highlight data can
+                have on the primary image data. This will be multiplied by
+                the linearly scaled highlight data and then added or
+                subtracted from the highlight channels. See class docstring
+                for more information.
+            highlight_channel (tuple): Series of booleans or None for every
+                channel in the RGBA image (4). True means apply the highlight
+                effect by adding, False means apply the highlight effect by
+                subtracting, and None means don't apply the highlight. By
+                default this will add to the Red and Green channels, subtract
+                from the Blue channel, and not effect the Alpha channel. This
+                results in yellow highlights in the resulting image.
+
+        """
+        self.min_highlight = min_highlight
+        self.max_highlight = max_highlight
+        self.max_factor = max_factor
+        self.highlight_channel = highlight_channel
+        super().__init__(name, **kwargs)
+
+    def _get_highlight_factor(self, highlight_data):
+        factor = (highlight_data - self.min_highlight) / (self.max_highlight - self.min_highlight)
+        factor = factor.where(factor.notnull(), 0) * self.max_factor
+        return factor
+
+    def _apply_highlight_effect(self, background_data, factor):
+        new_channels = []
+        for highlight_effect, band_name in zip(self.highlight_channel, "RGBA"):
+            if highlight_effect:
+                channel_factor = factor
+            elif highlight_effect is None:
+                channel_factor = None
+            else:
+                channel_factor = -factor
+
+            new_channel = background_data.sel(bands=[band_name])
+            if channel_factor is not None:
+                new_channel = new_channel + channel_factor
+            new_channels.append(new_channel)
+        return new_channels
+
+    def __call__(self, projectables, optional_datasets=None, **attrs):
+        """Create B/W image with highlighted pixels."""
```

Comment:
If this works only with B/W, should it raise an exception if it's colour?

Reply:
Hhhhmmm, it converts whatever is provided to a RGBA image. Technically it would work fine if provided a background image that is already RGB but I'm not sure the use case. The B/W part of this docstring is just that it assumes that the enhanced version of the background image will be B/W. I think I'll update the docstring rather than limit the functionality of the compositor.